src/pybind/mgr/cephadm/services/nvmeof.py

```diff
@@ -212,6 +212,29 @@ def ok_to_stop(self,
         warn_message = f'It is presumed safe to stop {names}'
         return HandleCommandResult(0, warn_message, '')
 
+    def ignore_possible_stray(
+        self, service_type: str, daemon_id: str, name: str
+    ) -> bool:
+        if service_type == 'nvmeof':
+            # Some newer versions of nvmeof will register with the cluster
+            # with a name that does not include the pool or group name
+            # getting us from "nvmeof.<pool>.<group>.<hostname>.<6-random-chars>"
+            # to "nvmeof.<hostname>.<6-random-chars>"
+            #
+            # This isn't a perfect solution, but we're assuming here if the
+            # random chars at the end of the daemon name match a daemon
+            # we know, it's likely not a stray
+            try:
+                random_chars = daemon_id.split('.')[-1]
+            except ValueError:
+                logger.debug('got nvmeof daemon id: "%s" with no dots', daemon_id)
```

Comment:
should this be an error as it should happen in production?

Reply:
This is the daemon id from list_servers which we don't control. I just noticed the stray daemon warnings appear in a test run after things were changed elsewhere, so I'm a bit worried about throwing an exception here. I wouldn't want the serve loop getting stuck at `_check_daemons`. I am willing to raise the log level of this though at least